rfcs/models/identity/identity-update-policy.md

```diff
@@ -0,0 +1,69 @@
+
+`updatePolicy` defines a combination of participant units that can sign and authorize a Network DID updation request. This `updatePolicy` has the condition format as specified in [`verifiablecondition2021`](https://w3c.github.io/did-spec-registries/#verifiablecondition2021) verification method. It can be used to combine different verification methods together to create complex conditional expressions such as logical operations, thresholds, weighted thresholds, relationships and a delegation to external verification methods.
+
+
+Example of `updatePolicy`. 
+```json
+{
+  "id": "did:<iin_name>:<network_name>#updatepolicy",
+  "controller": "did:<iin_name>:<network_name>",
+  "type": "VerifiableCondition2021",
+  "conditionAnd": [{
+      "id": "did:<iin_name>:<network_name>#updatepolicy-1",
+      "controller": "did:<iin_name>:<network_name>",
+      "type": "VerifiableCondition2021",
+      "conditionOr": ["did:<iin_name>:<network_participant_3>#key1",
+        "did:<iin_name>:<network_participant_2>#key3"
+      ]
+    },
+    "did:<iin_name>:<network_participant_1>#key1"
+  ]
+}
+```
+
+## Network DID updates
+
+Any request to update the *Network DID* must be authenticated based on different policies for the different scenarios of (i) participants leaving the network, (ii) new participants joining the network, and (iii) other updates to the Network DID.
+
+### Participants leaving the network
+
+There can be two scenarios when an existing participant unit of a network leaves: (a) The participant itself willingly leaves the network. (b) Other participants of the network decide to expel that participant. Based on these two scenarios, any *Network DID* update request removing one participant from the `networkParticipants` property is authenticated as per the following conditions:
+
+A. The *Network DID* update request is attested by the participant who is leaving the network, that is the participant whose DID is being excluded from the  `networkParticipants` property. This indicates voluntary exit from the network.
+
+B. If condition A is not satisfied, and there exists an `updatePolicy` property of the *BlockchainNetworkMultiSig* `verificationMethod` in the *Network DID*, then the DID update request must be attested satisfying the `updatePolicy`.
+
+C. If condition A is not satisfied and there is no `updatePolicy` property of the *BlockchainNetworkMultiSig* `verificationMethod` in the *Network DID*, then the DID update request must be attested by strictly greater than half (`> N/3`) of the existing participants in the network indicated by `networkParticipants` property of the existing *Network DID* document.
```

Comment:
I am not sure about having such a default behaviour specified. It feels like it might open up security issues. The safest assumption here might be that if a network does not specify and update policy, we assume it means for its member set to be static and immutable. If it desires anything else, it should explicitly say so by defining an update policy (provided the update policy offers a way to also express thresholds 1/3, 1/2, 2/3 N etc.)

Reply:
In the updated version I have removed this default behavior. Instead i have included this part under "Choice of updatePolicy":


During the creation of a *Network DID*, if no `updatePolicy` is set in the Network DID Document, then a default `updatePolicy` may be set by the IIN as a threshold condition requiring the attestation of `> N/2` network participants.

Example:

```json
{
  "id": "did:<iin_name>:<network_name>#updatepolicy",
  "controller": "did:<iin_name>:<network_name>",
  "type": "VerifiableCondition2021",
  "threshold": 4,
  "conditionThreshold": [
    "did:<iin_name>:<network_participant_1>#key1",
    "did:<iin_name>:<network_participant_2>#key3",
    "did:<iin_name>:<network_participant_3>#key2",
    "did:<iin_name>:<network_participant_4>#key1",
    "did:<iin_name>:<network_participant_5>#key1",
    "did:<iin_name>:<network_participant_6>#key1",
    "did:<iin_name>:<network_participant_7>#key1"
  ]
}
```

If the network members explicitely wants to create a "static" network where the network members are not supposed to be changed, then the threshold is set to a value more than the number of participants in the network.

Reply:
Let me know if this is better or should I just remove this N/2 part completely by only stating that threshold policites are possible.

Reply:
I'm fine with the above, but can we call it `defaultThreshold` just to make it clear?

Reply:
in the above example, the specified threshold is the number of parties that can approve?  should this be a percentage instead? since otherwise, that threshold number becomes outdated every time we add/remove a party?

Reply:
looking at https://w3c-ccg.github.io/verifiable-conditions/  I gather we don't have much control over this.  I guess our assumption here can be that if a network requires a percentage threshold to hold they have to continue to evolve this threshold number as members join/leave.